### PR TITLE
fix: Folder Sequencing When we try to move any folder at the bottom of the folders list

### PR DESCRIPTION
**Pull request by ravindra-bruno**
**JIRA: BRU-1112**

#### Contribution Checklist:

- [x] **I've used AI significantly to create this pull request**
- [x] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes**
- [ ] **I have added screenshots or gifs to help explain the change if applicable.**
- [ ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

- **New Features**
  - Refined collection drag-and-drop using consistent **above / inside / below** placement and matching visual drop highlighting.
  - Improved item move behavior with more accurate placement-aware reordering and pathname updates during cross-collection and cross-folder moves.

- **Bug Fixes**
  - Prevented invalid drops (onto itself or into a descendant path) and tightened drop validation to account for collection and ancestry rules.
  - Fixed edge cases to keep ordering stable when reordering folders alongside files.

- **Tests**
  - Added Jest coverage for drop-position resolution, drop permission rules, pathname updates, and reordering outcomes.

### CODE REVIEW

**Review by utkarsh-bruno — DISMISSED** [review]

File: packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js [hunk 1]
```diff
@@ -1267,7 +1268,7 @@ export const handleCollectionItemDrop
         }
 
         // Update sequences in the target directory (if dropping adjacent)
```
Comment (utkarsh-bruno) [inline packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js]: If the droptype is removed from adjacent and is now above or below, then change the comments to reflect the same

**Review by utkarsh-bruno — APPROVED** [review]
